waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingService.java

```diff
@@ -155,6 +162,18 @@ private void validateFederatedMetastoreDatabases(List<String> mappableDatabases,
     }
   }
 
+  private List<String> getDatabasesFromPattern(Iface client, List<String> mappedDatabases) {
+    List<String> matchedDatabases = new ArrayList<String>();
+    for (String mappedDatabase : mappedDatabases) {
+      try {
+        matchedDatabases.addAll(client.get_databases(mappedDatabase));
```

Comment:
I think the logic here needs to change. 
* We're doing a fetch for every "pattern" which is not very efficient.
* Some pattern might return duplicates on subsequent fetches
* The pattern allowed by the `client.get_database` call is I think something like: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ShowDatabases. So almost regex but not completely. 
I think it is better if we use java regex to compile the patterns. I think we do this in other places already. Then we can do a `client.get_all_databases` and match the results with our patterns. That will be only one client call.
I think the Integration test succeeds cause the pattern matches what hive supports a more complex regex will probably fail.